dae/dae/common_reports/common_report.py

```diff
@@ -161,3 +163,42 @@ def to_dict(self, full=False):
             "transmitted": self.transmitted,
             "study_description": self.study_description,
         }
+
+    def save(self, report_filename: str) -> None:
+        """Save common report into a file."""
+        if not os.path.exists(os.path.dirname(report_filename)):
+            os.makedirs(os.path.dirname(report_filename))
+        with open(report_filename, "w+", encoding="utf8") as crf:
+            json.dump(self.to_dict(full=True), crf)
+
+    @staticmethod
+    def load(report_filename: str) -> Optional[CommonReport]:
+        """Load a common report from a file.
+
+        I file does not exists returns None.
+        """
+        if not os.path.exists(report_filename):
+            return None
+        with open(report_filename, "r", encoding="utf-8") as crf:
+            cr_json = json.load(crf)
+
+        return CommonReport(cr_json)
+
+    @staticmethod
+    def build_and_save(study, force=False):
+        """Build a common report for a study and saves it.
+
+        If the common reports and disabled for the study, the function skips
+        building the report and returns None.
+
+        If the report already exists the default behavior is to skip building
+        the report and return None. You can force building the report by
+        passing `force=True` to the function.
+        """
+        if not study.config.common_report.enabled:
+            return None
+        if os.path.exists(study.config.common_report.file_path) and not force:
+            return None
```

Comment:
Seems counter intuitive to me to return None when a common report is already generated. Is this for a specific use case?

Reply:
I changed it to load and return the report. This looks more consistent.